src/SFA.DAS.AssessorService.Application/Handlers/Certificates/UpdateCertificatesPrintStatusHandler.cs

```diff
@@ -28,14 +28,31 @@ public UpdateCertificatesPrintStatusHandler(ICertificateRepository certificateRe
 
         public async Task<ValidationResponse> Handle(UpdateCertificatesPrintStatusRequest request, CancellationToken cancellationToken)
         {
-            var validationResult = new ValidationResponse();
-
+            var validationResult = new ValidationResponse();            
+          
             var validatedCertificatePrintStatuses = await Validate(request.CertificatePrintStatuses, validationResult);
             foreach(var validatedCertificatePrintStatus in validatedCertificatePrintStatuses)
-            {
+            {                
+                var certificateBatchLog = await _certificateRepository.GetCertificateBatchLog(validatedCertificatePrintStatus.CertificateReference, validatedCertificatePrintStatus.BatchNumber);
+                if (certificateBatchLog == null)
+                {
+                    //Scenario 3:  Certificate not printed but included in delivery notification file
+                    validationResult.Errors.Add(
+                      new ValidationErrorDetail("CertificatePrintStatuses", $"Certificate {validatedCertificatePrintStatus.CertificateReference} not printed in batch {validatedCertificatePrintStatus.BatchNumber}  .", ValidationStatusCode.NotFound));                    
+                }
+                else
+                {
+                    //Scenario 5:  StatusChangedDateTime earlier than the latest date of the previous status for a certificate number in the given batch for eg delivery date earlier than the printed date
+                    if (validatedCertificatePrintStatus.StatusChangedAt < certificateBatchLog.LatestChange())
```

Comment:
This should be certificateBatchLog.StatusAt rather than LatestChange, the difference being when was the status changed rather than when did we know it was changed

Reply:
done